### PR TITLE
Subject Request Events and Logs Section

### LINKED ISSUE #495: [Admin UI] Subject Request details - Activity and Event logs
### Acceptance Criteria

1. Display order of events in chronological order (most recent on top aka descending), specifically the start time of the event.
- second ordering can be alphabetic IF start times are the exact same time.
- _note: andrew to look into this as there might be different options available for this sort depending on audit vs. execution logs._
2. For each “event” the following information MUST be displayed
- Event Title
- This is something that succinctly describes the type of activity (see table below)
- timestamp of when the event started
- Ability to “View Details”

Activity/Event Types | Event Title Format | “View Detail” information
-- | -- | --
Request was received by fidesops | Request received | Event Title“This request was received and awaits processing”Timestamp
Request was approved by a fidesops user | Request approved | Event Title“< username > approved this request” Timestamp
Request is being processed through a datastore | Processing < Datastore Name > Connector | Details should include when the datastore started and finished processingID of datastore?Datastore connection statusAny relevant informational logs (like the API response for SaaS connectors)
Request failed to process successfully through a datastore | Request errored at < Datastore Name > Connector | ID of datastore?timestamp of when the connector started timestamp of failureerror log of failure
Request was denied | Request denied | TimestampDenial reason
Fidesops finished processing the request | Request completed | Timestampis there a way to associate this request to a location within the connected s3 bucket?

### Design Artifacts
- [Mockup](https://www.figma.com/file/MTbZohIGvB4bq2f2wkiDHO/Fidesops-Management-UI?node-id=1142%3A12922) for viewing details on an event in the activity timeline
- [Mockup](https://www.figma.com/file/MTbZohIGvB4bq2f2wkiDHO/Fidesops-Management-UI?node-id=0%3A1) for SR detail page
- related design ticket - [FPD-13](https://ethyca.atlassian.net/browse/FPD-13)
- look at this loom [on this page](https://ethyca.atlassian.net/wiki/spaces/PM/pages/2296872986/Subject+Request+Detail+Pages#Processing-Activity%2FData)

Aha! Link: https://ethyca.aha.io/features/OPS-60

### PR DESCRIPTION
**Pull request by TheAndrewJackson**
# Purpose
Add Events and Logs to the Subject Request Details page. This PR implements everything that's possible right now. In order to finish out the whole issue the backend needs to add some additional logs for the activity timeline. This is a stop gap to merge in what's currently working. Once the backend has added everything that's needed a followup PR will finish this feature.

# Changes
- Add Events and Logs section to the subject request details page
- Add slide out drawer to view dataset collection logs and errors.

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #495

### CODE REVIEW

**Comment by seanpreston** [comment]
Hey @TheAndrewJackson, what data have you been using to test this? Did you write a script or are you running requests through and using the logs? Just thinking a script might help because we can be sure it covers every edge case.

**Comment by TheAndrewJackson** [comment]
I've been manually creating, or mocking, some of the data on the frontend. I created the execution logs by following the "Minimum API calls to create an Access Privacy Request" directory in our Postman collection. 

I agree that a script would be really helpful. Maybe the `test-data` script could be updated to add logs to each example subject request.

File: clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+  const closeDrawer = () => {
+    if (isViewingError) {
+      closeErrorPanel();
+    }
+
+    onClose();
+  };
+
+  const headerText = isViewingError ? "Event detail" : "Event log";
```
Comment (seanpreston) [inline clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx:54]: Did we think about having this popover update the URL? I could imagine a user wanting to share event logs for request failures for instance, as that's where the trace will be.

**Comment by seanpreston** [comment]
> I agree that a script would be really helpful. Maybe the test-data script could be updated to add logs to each example subject request.

We'd need to actually process the requests created, which means we'll need to run the server (rather than just access the DB directly). I'll have a play around this morning and see if that's an easy enough restructure.

More broadly we need to get better at planning test data in for these frontend features.

File: clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx (hunk 1 above)
Comment (TheAndrewJackson) [reply clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx:54]: I don't think so. That's a good idea but it won't work right now because our current BE router isn't able to resolve dynamic urls. Once we decide on a solution for dynamic urls we will be able support features like this.

Are you okay with adding this in a future PR once that BE supports this?

Comment (seanpreston) [reply clients/ops/admin-ui/src/features/subject-request/events-and-logs/EventDetails.tsx:54]: Sure! Have made a [ticket](https://github.com/ethyca/fidesops/issues/1027)

**Review by seanpreston — APPROVED** [review]
